src/libraries/icubmod/embObjMotionControl/eomcParser.h

```diff
@@ -56,6 +57,226 @@ namespace yarp {
+
+    virtual double convertRawToTempCelsius(const double temperature) = 0;
+
+    virtual motor_temperatureSensorTypeEnum_t getType() = 0;
+};
+
+class TemperatureSensorPT100 : public ITemperatureSensor
+{
+private:
+    // resistors of the voltage divider bridge
+    int _r_1;
+    int _r_2;
+    int _r_3;
+    
+
+    double _ptc_offset;    // offset of the temperature sensor line
+    double _ptc_gradient;  // slope/gradient of the temperature sensor line
+    double _pga_gain;      // ADC gain set for the tdb (temperature detection board)
+
+    int _vcc;               // vcc that enters the voltage divider bridge
+    double _resolution_pga; // resolution of the internal pga of the tdb
+    double _resolution_tdb; // resolution used for the raw value for the output of the tdb
+
+    double _half_bridge_resistor_coeff;
+
+    bool isConfigured = false;
+
+public:
+
+    TemperatureSensorPT100()
+    {
+        _r_1 = 4700;
+        _r_2 = 4700;
+        _r_3 = 100; 
+
+        _ptc_offset = 100.0;
+        _ptc_gradient = 0.3851;
+        _pga_gain = 2;
+        _vcc = 5;
+        _resolution_pga = 2.048;
+        _resolution_tdb = 32767; 
+
+        _half_bridge_resistor_coeff = (double)_r_3 / (double)(_r_2 + _r_3);
+        isConfigured = true;
+    }
+
+    TemperatureSensorPT100(const TemperatureSensorPT100& other) = default; // const copy constructor
+    TemperatureSensorPT100& operator=(const TemperatureSensorPT100& other) = default; // const copy assignment operator
+    TemperatureSensorPT100(TemperatureSensorPT100&& other) = default; // move constructor
+    TemperatureSensorPT100& operator=(TemperatureSensorPT100&& other) = default; // move assignment operator
+
+    ~TemperatureSensorPT100() = default; // destructor
+
+    virtual double convertTempCelsiusToRaw(const double temperature) override
+    {
+        double res = 0;
+        if (!isConfigured)
+        {
+            yError("Cannot proceed, class paramters not confgured");
+            return res;
+        }
+       	
+        double tmp = (( (_ptc_offset + _ptc_gradient * temperature) / ((double)_r_1 + (_ptc_offset + _ptc_gradient * temperature))) - _half_bridge_resistor_coeff) * (double)_vcc;
+        res = (_resolution_tdb + 1) * ((_pga_gain * tmp) / _resolution_pga);
+        
+	    yDebug("Converted temperature limit to raw value:%f", res);
+        return res;
+    }
+
+    virtual double convertRawToTempCelsius(const double temperature) override
+    {
+        double res = 0;
+        if (!isConfigured)
+        {
+            yError("Cannot proceed, class paramters not confgured");
+            return res;
+        }
+        
+        double tmp = temperature * ((_resolution_pga) / (_pga_gain * _vcc * (_resolution_tdb + 1)));
+        double den = _ptc_gradient * (_r_2 - _r_2*tmp - _r_3*tmp);
+        res = (tmp * (_r_1*_r_2 + _r_1*_r_3 + _ptc_offset*_r_2 + _ptc_offset*_r_3) / den) + ((_r_3*_r_1 - _r_2*_ptc_offset) / den);
```

Comment:
Please put `(_r_1*_r_2 + _r_1*_r_3 + _ptc_offset*_r_2 + _ptc_offset*_r_3) / den) + ((_r_3*_r_1 - _r_2*_ptc_offset) / den`in a const field. This small change speeds up a bit the calculation. 

Same for PT1000

Reply:
Check last commit. Update as requested